### ISSUE TITLE
Unable to bootstrap Vault

### ISSUE THREAD

**Issue by galvesribeiro** [hashicorp/vault#9903]
I'm just trying to bootstrap Vault (non-enterprise) on Kubernetes in Azure AKS. After followed the docs and adapting it to our infrastructure we've came out with this `values.yaml`:

```
certName: pl-vault-dev
vault:
  global:
    enabled: true
    tlsDisable: false

  injector:
    enabled: false
    image:
      repository: "hashicorp/vault-k8s"
      tag: "1.5.3"

    resources:
        requests:
          memory: 256Mi
          cpu: 250m
        limits:
          memory: 256Mi
          cpu: 250m
  server:
    image:
      repository: "hashicorp/vault"
      tag: "1.5.3"
    resources:
      requests:
        memory: 8Gi
        cpu: 2000m
      limits:
        memory: 8Gi
        cpu: 2000m  
    ingress:
      enabled: true
      annotations:
        kubernetes.io/ingress.class: azure/application-gateway
        appgw.ingress.kubernetes.io/ssl-redirect: "true"
        appgw.ingress.kubernetes.io/backend-protocol: "https"
        appgw.ingress.kubernetes.io/backend-hostname: vault-dev.mydomain.com.br
        appgw.ingress.kubernetes.io/appgw-trusted-root-certificate: pl-ca
        appgw.ingress.kubernetes.io/request-timeout: "360"
      hosts:
        - host: vault-dev.mydomain.com.br
          paths:
            - /
            - /vault
      tls:
        - hosts:
            - vault-dev.mydomain.com.br
          secretName: pagolivre-tls
    readinessProbe:
      enabled: true
      path: "/v1/sys/health?standbyok=true&sealedcode=204&uninitcode=204"
    livenessProbe:
      enabled: true
      path: "/v1/sys/health?standbyok=true"
      initialDelaySeconds: 60
    service:
      annotations:
        service.beta.kubernetes.io/azure-load-balancer-internal: "true"
        service.beta.kubernetes.io/azure-load-balancer-internal-subnet: PLClusterServicesSubnet
    extraEnvironmentVars:
      VAULT_CACERT: /app/certificates/ca.crt
    volumes:
      - name: certificate
        secret:
          secretName: pl-vault-dev-certificate
    volumeMounts:
      - mountPath: "/app/certificates/"
        name: certificate
    auditStorage:
      enabled: true
      size: 10Gi
      storageClass: managed-retain
      accessMode: ReadWriteOnce
    dataStorage:
      enabled: true
      size: 10Gi
      storageClass: managed-retain
      accessMode: ReadWriteOnce
    standalone:
      enabled: false
    ha:
      enabled: true
      replicas: 2
      raft:
        enabled: true
        setNodeId: true
        config: |
          ui = true
          listener "tcp" {
            address = "0.0.0.0:8200"
            cluster_address = "0.0.0.0:8201"
            tls_cert_file = "/app/certificates/tls.crt"
            tls_key_file = "/app/certificates/tls.key"
            tls_ca_cert_file = "/app/certificates/ca.crt"
          }

          storage "raft" {
            path = "/vault/data"
            retry_join {
              leader_api_addr = "https://pl-vault-0:8200"
              leader_ca_cert_file = "/app/certificates/ca.crt"
              leader_client_cert_file = "/app/certificates/tls.crt"
              leader_client_key_file = "/app/certificates/tls.key"
            }
            retry_join {
              leader_api_addr = "https://pl-vault-1:8200"
              leader_ca_cert_file = "/app/certificates/ca.crt"
              leader_client_cert_file = "/app/certificates/tls.crt"
              leader_client_key_file = "/app/certificates/tls.key"
            }
          }

          service_registration "kubernetes" {}

  ui:
    enabled: true
    serviceType: "LoadBalancer"
    serviceNodePort: null
    externalPort: 8200
    annotations:
      service.beta.kubernetes.io/azure-load-balancer-internal: "true"
      service.beta.kubernetes.io/azure-load-balancer-internal-subnet: PLClusterServicesSubnet
```

After `helm install`, I can see the pods being running with the PVC volumes mounted, the certificate is also mounted (provided by Cert-Manager) etc.

However, it fails when I try to initiate it as suggested by the documentation:

```
kubectl exec -ti -n vault-dev pl-vault-0 -- vault operator init
Error initializing: Put "https://127.0.0.1:8200/v1/sys/init": net/http: TLS handshake timeout
command terminated with exit code 2
```

The pods are in Running state but keep restarting:

```
kubectl get pods -l app.kubernetes.io/name=vault -n vault-dev
NAME         READY   STATUS    RESTARTS   AGE
pl-vault-0   0/1     Running   1          4m2s
pl-vault-1   0/1     Running   1          4m2s
```

And on every restart I have this log:

```
vault
==> Vault server configuration:
vault
Api Address: https://100.64.1.72:8200
vault
Cgo: disabled
vault
Cluster Address: https://pl-vault-0.pl-vault-internal:8201
vault
Go Version: go1.14.7
vault
Listener 1: tcp (addr: "0.0.0.0:8200", cluster address: "0.0.0.0:8201", max_request_duration: "1m30s", max_request_size: "33554432", tls: "enabled")
vault
Log Level: info
vault
Mlock: supported: true, enabled: false
vault
Recovery Mode: false
vault
Storage: raft (HA available)
vault
Version: Vault v1.5.3
vault
Version Sha: 9fcd81405feb320390b9d71e15a691c3bc1daeef
vault
==> Vault server started! Log data will stream in below:
vault
==> Vault shutdown triggered
vault
2020-09-05T23:01:25.314Z [INFO] proxy environment: http_proxy= https_proxy= no_proxy=
vault
2020-09-05T23:01:25.406Z [INFO] core: raft retry join initiated
vault
2020-09-05T23:01:25.406Z [INFO] core: security barrier not initialized
vault
2020-09-05T23:01:25.406Z [INFO] core: security barrier not initialized
vault
2020-09-05T23:01:25.406Z [INFO] core: attempting to join possible raft leader node: leader_addr=https://pl-vault-0:8200
vault
2020-09-05T23:01:35.415Z [INFO] core: join attempt failed: error="error during raft bootstrap init call: Put "https://pl-vault-0:8200/v1/sys/storage/raft/bootstrap/challenge": net/http: TLS handshake timeout"
vault
2020-09-05T23:01:35.415Z [INFO] core: security barrier not initialized
vault
2020-09-05T23:01:35.415Z [INFO] core: attempting to join possible raft leader node: leader_addr=https://pl-vault-1:8200
vault
2020-09-05T23:01:35.419Z [INFO] core: join attempt failed: error="error during raft bootstrap init call: Put "https://pl-vault-1:8200/v1/sys/storage/raft/bootstrap/challenge": dial tcp: lookup pl-vault-1 on 192.168.0.10:53: no such host"
vault
2020-09-05T23:01:35.419Z [ERROR] core: failed to retry join raft cluster: retry=2s
vault
2020-09-05T23:01:37.419Z [INFO] core: security barrier not initialized
vault
2020-09-05T23:01:37.419Z [INFO] core: attempting to join possible raft leader node: leader_addr=https://pl-vault-0:8200
vault
2020-09-05T23:01:47.422Z [INFO] core: join attempt failed: error="error during raft bootstrap init call: Put "https://pl-vault-0:8200/v1/sys/storage/raft/bootstrap/challenge": net/http: TLS handshake timeout"
vault
2020-09-05T23:01:47.422Z [INFO] core: security barrier not initialized
vault
2020-09-05T23:01:47.422Z [INFO] core: attempting to join possible raft leader node: leader_addr=https://pl-vault-1:8200
vault
2020-09-05T23:01:47.425Z [INFO] core: join attempt failed: error="error during raft bootstrap init call: Put "https://pl-vault-1:8200/v1/sys/storage/raft/bootstrap/challenge": dial tcp: lookup pl-vault-1 on 192.168.0.10:53: no such host"
vault
2020-09-05T23:01:47.425Z [ERROR] core: failed to retry join raft cluster: retry=2s
vault
2020-09-05T23:01:49.426Z [INFO] core: security barrier not initialized
vault
2020-09-05T23:01:49.426Z [INFO] core: attempting to join possible raft leader node: leader_addr=https://pl-vault-0:8200
vault
2020-09-05T23:01:59.429Z [INFO] core: join attempt failed: error="error during raft bootstrap init call: Put "https://pl-vault-0:8200/v1/sys/storage/raft/bootstrap/challenge": net/http: TLS handshake timeout"
vault
2020-09-05T23:01:59.430Z [INFO] core: security barrier not initialized
vault
2020-09-05T23:01:59.430Z [INFO] core: attempting to join possible raft leader node: leader_addr=https://pl-vault-1:8200
vault
2020-09-05T23:01:59.435Z [INFO] core: join attempt failed: error="error during raft bootstrap init call: Put "https://pl-vault-1:8200/v1/sys/storage/raft/bootstrap/challenge": dial tcp: lookup pl-vault-1 on 192.168.0.10:53: no such host"
vault
2020-09-05T23:01:59.435Z [ERROR] core: failed to retry join raft cluster: retry=2s
vault
2020-09-05T23:02:01.435Z [INFO] core: security barrier not initialized
vault
2020-09-05T23:02:01.435Z [INFO] core: attempting to join possible raft leader node: leader_addr=https://pl-vault-0:8200
vault
2020-09-05T23:02:11.443Z [INFO] core: join attempt failed: error="error during raft bootstrap init call: Put "https://pl-vault-0:8200/v1/sys/storage/raft/bootstrap/challenge": net/http: TLS handshake timeout"
vault
2020-09-05T23:02:11.443Z [INFO] core: security barrier not initialized
vault
2020-09-05T23:02:11.443Z [INFO] core: attempting to join possible raft leader node: leader_addr=https://pl-vault-1:8200
vault
2020-09-05T23:02:11.448Z [INFO] core: join attempt failed: error="error during raft bootstrap init call: Put "https://pl-vault-1:8200/v1/sys/storage/raft/bootstrap/challenge": dial tcp: lookup pl-vault-1 on 192.168.0.10:53: no such host"
vault
2020-09-05T23:02:11.448Z [ERROR] core: failed to retry join raft cluster: retry=2s
vault
2020-09-05T23:02:13.449Z [INFO] core: security barrier not initialized
vault
2020-09-05T23:02:13.449Z [INFO] core: attempting to join possible raft leader node: leader_addr=https://pl-vault-0:8200
vault
2020-09-05T23:02:23.507Z [INFO] core: join attempt failed: error="error during raft bootstrap init call: Put "https://pl-vault-0:8200/v1/sys/storage/raft/bootstrap/challenge": net/http: TLS handshake timeout"
vault
2020-09-05T23:02:23.507Z [INFO] core: security barrier not initialized
vault
2020-09-05T23:02:23.507Z [INFO] core: attempting to join possible raft leader node: leader_addr=https://pl-vault-1:8200
vault
2020-09-05T23:02:23.513Z [INFO] core: join attempt failed: error="error during raft bootstrap init call: Put "https://pl-vault-1:8200/v1/sys/storage/raft/bootstrap/challenge": dial tcp: lookup pl-vault-1 on 192.168.0.10:53: no such host"
vault
2020-09-05T23:02:23.513Z [ERROR] core: failed to retry join raft cluster: retry=2s
vault
2020-09-05T23:02:25.513Z [INFO] core: security barrier not initialized
vault
2020-09-05T23:02:25.513Z [INFO] core: attempting to join possible raft leader node: leader_addr=https://pl-vault-0:8200
vault
2020-09-05T23:02:35.517Z [INFO] core: join attempt failed: error="error during raft bootstrap init call: Put "https://pl-vault-0:8200/v1/sys/storage/raft/bootstrap/challenge": net/http: TLS handshake timeout"
vault
2020-09-05T23:02:35.517Z [INFO] core: security barrier not initialized
vault
2020-09-05T23:02:35.517Z [INFO] core: attempting to join possible raft leader node: leader_addr=https://pl-vault-1:8200
vault
2020-09-05T23:02:35.522Z [INFO] core: join attempt failed: error="error during raft bootstrap init call: Put "https://pl-vault-1:8200/v1/sys/storage/raft/bootstrap/challenge": dial tcp: lookup pl-vault-1 on 192.168.0.10:53: no such host"
vault
2020-09-05T23:02:35.522Z [ERROR] core: failed to retry join raft cluster: retry=2s
vault
2020-09-05T23:02:37.522Z [INFO] core: security barrier not initialized
vault
2020-09-05T23:02:37.522Z [INFO] core: attempting to join possible raft leader node: leader_addr=https://pl-vault-0:8200
vault
2020-09-05T23:02:46.351Z [WARN] service_registration.kubernetes: unable to set initial state due to context canceled, will retry
vault
2020-09-05T23:02:46.351Z [WARN] service_registration.kubernetes: unable to set initial state due to context canceled, will retry
```

Can anyone point us to the right direction?

Thank you!

**Comment by galvesribeiro** [comment]
I gave up mostly. The support is without paying the enterprise license is really hard to find. There is not a really big community using it openly or they're not allowed to speak more.

In the end of the day, I got it working in a test cluster IIRC by using cert-manager using the IP with the correct SAN. 

Its being a while and we didn't decide to move forward with it to production.

Hopefully You have a bit more luck than myself.
